akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -50,15 +51,17 @@ private[http] object StreamUtils {
     impl.ErrorPublisher(ex, "failed").asInstanceOf[Publisher[T]]
 
   def mapErrorTransformer(f: Throwable ⇒ Throwable): Flow[ByteString, ByteString, NotUsed] = {
-    val transformer = new PushStage[ByteString, ByteString] {
-      override def onPush(element: ByteString, ctx: Context[ByteString]): SyncDirective =
-        ctx.push(element)
+    val transformer = new SimpleLinearGraphStage[ByteString] {
+      override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler with OutHandler {
+        override def onPush(): Unit = push(out, grab(in))
+        override def onUpstreamFailure(ex: Throwable): Unit = failStage(f(ex))
+        override def onPull(): Unit = pull(in)
 
-      override def onUpstreamFailure(cause: Throwable, ctx: Context[ByteString]): TerminationDirective =
-        ctx.fail(f(cause))
+        setHandlers(in, out, this)
+      }
     }
 
-    Flow[ByteString].transform(() ⇒ transformer).named("transformError")
+    Flow[ByteString].via(transformer).named("transformError")
```

Comment:
I just went with a literal translation, probably worth looking at a refactor of the callers though;

`CodingDirectives.decodeRequestWith` looks like the only user.

    effectiveDecoder.decode(request).mapEntity(StreamUtils.mapEntityError {
              case NonFatal(e) ⇒
                IllegalRequestException(
                  StatusCodes.BadRequest,
                  ErrorInfo("The request's encoding is corrupt", e.getMessage))
            })

Perhaps just refactor to 

    effectiveDecoder.decode(request).mapEntity { entity ⇒
              entity.transformDataBytes(Flow[ByteString].recover {
                case NonFatal(e) ⇒
                  throw IllegalRequestException(
                    StatusCodes.BadRequest,
                    ErrorInfo("The request's encoding is corrupt", e.getMessage))
              })
            }```